### PR TITLE
Url regex matching

### PR DESCRIPTION
**Pull request by Mews**
This pr closes issue #13 

**Changes**
- Added `url_regex` argument to the `Spider` class.
- Urls then get matched against this pattern inside the `Spider.crawl` method. If they don't match the pattern, they are skipped.
- Also added a test case for this feature, in `test_url_regex`.

Please let me know if any changes need to be made to this pr.

### CODE REVIEW

File: tiny_web_crawler/crawler.py [hunk 1]
```diff
@@ -152,6 +155,11 @@ def crawl(self, url: str) -> None:
             if pretty_url in self.crawl_result[url]['urls']:
                 continue
 
+            if self.url_regex:
+                if not re.compile(self.url_regex).match(pretty_url):
+                    self.verbose_print(Fore.RED + f"Url didn't match regex: {pretty_url}")
```
Comment (indrajithi) [inline tiny_web_crawler/crawler.py:160]: We can remove this verbose_print or make it more meaningful to the user. Something like `"Skipping: URL didn't match regx: {pretty_url}"`. And we can keep it like a warning colour yellow. WDYT?

**Review by indrajithi — CHANGES_REQUESTED** [review]

File: tiny_web_crawler/crawler.py (hunk 1 above)
Comment (Mews) [reply tiny_web_crawler/crawler.py:160]: Yeah of course I'll change it right away 👍

Comment (Mews) [reply tiny_web_crawler/crawler.py]: Done

**Comment by indrajithi** [comment]
@Mews Can you make a pull from master. The CI fails because of some outdated workflow in master.

**Comment by Mews** [comment]
I don't think that was the issue, it was because of the `import re` line, it should be before requests

**Pull request merged**